python/src/iceberg/expressions/base.py

```diff
@@ -267,6 +204,34 @@ def __str__(self) -> str:
         return "false"
 
 
+class In(BooleanExpression):
```

Comment:
Now that we're moving away from having the single `Predicate` class, I still think we can do some things to allow grouping downstream (for like methods that must take a "predicate" or more specifically an "unbound predicate"). What do you think of adding this?

```py
class Predicate:

    @abstractmethod
    def __init__(self, left, right):
        """A concrete predicate must have a `left` and `right`"""

class Bound:
    pass

class Unbound:
    pass

class UnboundIn(BooleanExpression, Predicate, Unbound):
    def __init__(self, left: 'UnboundReference', right: List[Literal]):
        ...
```

Reply:
I've made some changes based on your suggestions and creating `UnboundIn`